plugin-code/src/main/java/com/trivago/features/FeatureFileConverter.java

```diff
@@ -317,14 +322,39 @@ private int generateRunnerClassesWithDesiredNumberOfRunners(final List<String> g
         }
 
         int currentRunnerIndex = 0;
+        int matchCount = 0;
         for (String generatedFeatureName : generatedFeatureNames) {
-            generatedFeatureNamesPerRunner.get(currentRunnerIndex).add(generatedFeatureName);
-            currentRunnerIndex++;
-            if (currentRunnerIndex >= targetRunnerNumber) {
-                currentRunnerIndex = 0;
+            if (scenarioNames.isEmpty()) {
+                generatedFeatureNamesPerRunner.get(currentRunnerIndex).add(generatedFeatureName);
+                currentRunnerIndex++;
+                if (currentRunnerIndex >= targetRunnerNumber) {
+                    currentRunnerIndex = 0;
+                }
+            } else {
+                // Move all scenarios matching a scenario name into its own group.
+                String scenarioText = fileIO.readContentFromFile(propertyManager.getGeneratedFeatureDirectory() + "/" + generatedFeatureName + ".feature");
+
+                if (scenarioText != null) {
+                    for (String scenarioName : scenarioNames) {
+                        String regex = "Scenario:.+" + scenarioName;
+                        Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+                        Matcher matcher = pattern.matcher(scenarioText);
+                        if (matcher.find()) {
+                            generatedFeatureNamesPerRunner.get(scenarioNames.indexOf(scenarioName)).add(generatedFeatureName);
+                            matchCount++;
+                            break;
+                        }
+                    }
+                }
             }
         }
 
+        if (!scenarioNames.isEmpty() && matchCount == 0) {
+            throw new CucablePluginException(
```

Comment:
It might be better to just log this instead of throwing an exception. Otherwise the plugin would just stop working.